sage-ios/SAGE/SAGE/LoginController.swift

```diff
@@ -66,16 +71,21 @@ class LoginController: UIViewController {
                     (valid: Bool) -> Void in
                     dispatch_async(dispatch_get_main_queue(), { () -> Void in
                         if (valid) {
-                            if let verified = LoginOperations.getUser()?.verified {
-                                if verified {
+                            LoginOperations.getState({ (user, currentSemester, userSemester) -> Void in
+                                KeychainWrapper.setObject(user, forKey: KeychainConstants.kUser)
+                                if !(currentSemester == nil) {
+                                    KeychainWrapper.setObject(currentSemester!, forKey: KeychainConstants.kCurrentSemester)
+                                }
+                                if !(userSemester == nil) {
+                                    KeychainWrapper.setObject(userSemester!, forKey: KeychainConstants.kSemesterSummary)
+                                }
+                                if (user.verified) {
                                     self.pushRootTabBarController()
                                 } else {
-                                    let unverifiedController = UnverifiedViewController()
-                                    self.presentViewController(unverifiedController, animated: true, completion: nil)
+                                    self.pushUnverifiedViewController()
                                 }
-                            } else {
-                                let unverifiedController = UnverifiedViewController()
-                                self.presentViewController(unverifiedController, animated: true, completion: nil)
+                                }) { (errorMessage) -> Void in
+                                    // error
```

Comment:
set an alertcontroller popup saying tehre was an error, and then log the person out like you would if the user pressed log out from profile